drivers/overlay/overlay.go

```diff
@@ -281,10 +278,26 @@ func supportsOverlay() error {
 			return nil
 		}
 	}
-	logrus.Error("'overlay' not found as a supported filesystem on this host. Please ensure kernel is new enough and has overlay support loaded.")
 	return errors.Wrap(graphdriver.ErrNotSupported, "'overlay' not found as a supported filesystem on this host. Please ensure kernel is new enough and has overlay support loaded.")
 }
 
+func supportsOverlay() error {
+	var err error
+	// We can try to modprobe overlay first before looking at
+	// proc/filesystems for when overlay is supported
+	exec.Command("modprobe", "overlay").Run()
+	for i := 0; i < 3; i++ {
```

Comment:
Might be good to leave a comment here on the race condition and the thinking for the retry.